### PR TITLE
Refactored readiness check handler and fixed metrics

### PR DESCRIPTION
**Pull request by sayanh**
**Description**

Changes proposed in this pull request:

- KnativeLib creates informers for Channel and readiness probe was using it. So for each readiness- check it was creating informers which were not used later. As a result, memory usage kept growing. Hence, removed the usage of KnativeLib from ReadinessProbe handler.
- Fixed the metrics for the published messages.

**Related issue(s)**

See #6299

### CODE REVIEW

File: components/event-bus/cmd/event-publish-service/publisher/publisher.go [hunk 1]
```diff
@@ -14,8 +14,10 @@ import (
 const (
 	// FAILED status label
 	FAILED = "failed"
-	// IGNORED status label
-	IGNORED = "ignored"
+	// IGNORED_NO_CHANNEL status label
+	IGNORED_NO_CHANNEL = "ignored_no_channel"
+	// IGNORED_NOT_READY_CHANNEL status label
+	IGNORED_NOT_READY_CHANNEL = "ignored_not_ready_channel"
```
Comment (k15r) [inline components/event-bus/cmd/event-publish-service/publisher/publisher.go]: can we make it sound more like natural language and less like :yoda:
`IGNORED_CHANNEL_NOT_READY`

File: components/event-bus/cmd/event-publish-service/publisher/publisher.go [hunk 2]
```diff
@@ -133,7 +125,17 @@ func (publisher *DefaultKnativePublisher) publishOnChannel(knativeLib *knative.K
 		log.Printf("failed to send message to the knative channel '%v' in namespace '%v'", channel.Name, *namespace)
 		return api.ErrorResponseInternalServer(), FAILED, channel.Name
 	}
+	report(channel, PUBLISHED, *namespace)
 
 	// publish to channel succeeded return nil error
 	return nil, PUBLISHED, channel.Name
 }
+
+func report(channel *messagingV1Alpha1.Channel, status, ns string) {
```
Comment (k15r) [inline components/event-bus/cmd/event-publish-service/publisher/publisher.go]: <nitpick>
something similar to `updateMetrics` would be more descriptive
</nitpick>

File: components/event-bus/cmd/event-publish-service/application/application.go [hunk 3]
```diff
@@ -56,7 +57,7 @@ func (app *KnativePublishApplication) start() {
 
 	// mark the app as started and register the readiness and the publish handlers
 	app.started = true
-	app.registerReadinessProbe()
+	app.registerReadinessProbe(app.knativeLib.EvClient())
```
Comment (k15r) [inline components/event-bus/cmd/event-publish-service/application/application.go]: A check should ensure that `knativeLib != nil`

Comment (sayanh) [reply components/event-bus/cmd/event-publish-service/application/application.go]: Added

File: components/event-bus/cmd/event-publish-service/publisher/publisher.go (hunk 2 above)
Comment (sayanh) [reply components/event-bus/cmd/event-publish-service/publisher/publisher.go]: Done

File: components/event-bus/cmd/event-publish-service/publisher/publisher.go (hunk 1 above)
Comment (sayanh) [reply components/event-bus/cmd/event-publish-service/publisher/publisher.go]: Done

File: components/event-bus/cmd/event-publish-service/application/application.go [hunk 4]
```diff
@@ -56,7 +57,10 @@ func (app *KnativePublishApplication) start() {
 
 	// mark the app as started and register the readiness and the publish handlers
 	app.started = true
-	app.registerReadinessProbe()
+	if app.knativeLib != nil {
```
Comment (antoineco) [inline components/event-bus/cmd/event-publish-service/application/application.go]: Shouldn't we just let it panic if nil? Doesn't sound like something we want to recover from.

File: components/event-bus/cmd/event-publish-service/application/application.go [hunk 5]
```diff
@@ -3,6 +3,7 @@ package application
 import (
 	"net/http"
 
+	messagingv1alpha1Client "github.com/knative/eventing/pkg/client/clientset/versioned/typed/messaging/v1alpha1"
```
Comment (antoineco) [inline components/event-bus/cmd/event-publish-service/application/application.go]: Capital letters in package aliases overlap with the variable names domain, better avoid them.
https://rakyll.org/style-packages/#lowercase-only

File: components/event-bus/internal/knative/util/kn-eventing.go [hunk 6]
```diff
@@ -203,6 +204,11 @@ func hasSynced(ctx context.Context, fn waitForCacheSyncFunc) error {
 	return nil
 }
 
+// MsgChannelClient returns a clientset interface for messaging v1alpha1 API
+func (k *KnativeLib) MsgChannelClient() messagingv1alpha1Client.MessagingV1alpha1Interface {
```
Comment (antoineco) [inline components/event-bus/internal/knative/util/kn-eventing.go]: Why "Channel"? Either return a "Messaging" `MessagingV1alpha1Interface` or a "Channel" `ChannelsGetter` if you are only interested about Channels.

Comment (sayanh) [reply components/event-bus/internal/knative/util/kn-eventing.go]: Unfortunately, "messagingChannel" is misleading. I renamed it to "msgClient" as I'm interested in the clientset for messagingv1alpha1(github.com/knative/eventing/pkg/client/clientset/versioned/typed/messaging/v1alpha1) API.

File: components/event-bus/cmd/event-publish-service/application/application.go (hunk 4 above)
Comment (sayanh) [reply components/event-bus/cmd/event-publish-service/application/application.go]: Alright, added a panic.

File: components/event-bus/internal/knative/util/kn-eventing.go (hunk 6 above)
Comment (antoineco) [reply components/event-bus/internal/knative/util/kn-eventing.go]: Sounds appropriate 👍

**Comment by sayanh** [comment]
Merging as we have a successful build https://status.build.kyma-project.io/log?id=1197902122599845888&job=pre-master-kyma-gke-upgrade

**Pull request merged**